Common/src/main/java/com/blamejared/crafttweaker/api/util/sequence/SequenceBuilder.java

```diff
@@ -0,0 +1,180 @@
+package com.blamejared.crafttweaker.api.util.sequence;
+
+import com.blamejared.crafttweaker.api.annotation.ZenRegister;
+import com.blamejared.crafttweaker.api.util.sequence.task.ISequenceTask;
+import com.blamejared.crafttweaker.api.util.sequence.task.type.InstantTask;
+import com.blamejared.crafttweaker.api.util.sequence.task.type.SleepTask;
+import com.blamejared.crafttweaker.api.util.sequence.task.type.SleepUntilTask;
+import com.blamejared.crafttweaker.impl.util.sequence.SequenceManager;
+import com.blamejared.crafttweaker.platform.sides.DistributionType;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import org.openzen.zencode.java.ZenCodeType;
+
+import java.util.LinkedList;
+import java.util.function.*;
+
+/**
+ * A builder for {@link Sequence}.
+ *
+ * @docParam this level.sequence()
+ */
+@ZenRegister
+@Document("vanilla/api/util/sequence/SequenceBuilder")
+@ZenCodeType.Name("crafttweaker.api.util.sequence.SequenceBuilder")
+public class SequenceBuilder<T, U> {
+    
+    
+    private final SequenceType type;
+    private final Supplier<T> actor;
+    private final U data;
+    private final boolean isClientSide;
+    private final LinkedList<ISequenceTask<T, U>> timeline;
+    
+    public SequenceBuilder(SequenceType type, Supplier<T> actor, U data, boolean isClientSide) {
```

Comment:
Don't get me wrong, I love `Supplier`s, but I'm curious. What's the rationale for it here? Mostly asking so there's a place we can look at in case we have this same question in the future.

Reply:
It was done so that if we have something like a Player sequence, where we may need to get a new player whenever we need it (if the player dies / changes dimension, it is a different Player entity).